python/pyspark/sql/context.py

```diff
@@ -463,6 +481,53 @@ def streams(self):
         return StreamingQueryManager(self._ssql_ctx.streams())
 
 
+class HiveContext(SQLContext):
+    """A variant of Spark SQL that integrates with data stored in Hive.
+
+    Configuration for Hive is read from ``hive-site.xml`` on the classpath.
+    It supports running both SQL and HiveQL commands.
+
+    :param sparkContext: The SparkContext to wrap.
+    :param jhiveContext: An optional JVM Scala HiveContext. If set, we do not instantiate a new
+        :class:`HiveContext` in the JVM, instead we make all calls to this object.
+
+    .. note:: Deprecated in 2.0.0. Use SparkSession.builder.enableHiveSupport().getOrCreate().
```

Comment:
Please rethink about this. Although the other removed API are recovered, I believe we should not recover this.
cc @marmbrus

Reply:
Can you justify this position using the rubric we agreed upon?

Reply:
I believe @gatorsmile should justify this using the new rubric. We are reviewing this PR.
Also, cc @rxin since he is the release manager.

Reply:
Look at the code in this PR. I see a bunch of alias methods that call other methods and a few minimal tests. What part of this is hard to maintain?

Reply:
`class HiveContext(SQLContext)` itself is not a simple alias. You know that. This is a whole new layer in Spark 1.6.x age.

Reply:
I see 63 lines of code in that file? That class itself looks like an alias... not a whole separate implementation. What am I missing?

Reply:
It seems that we have different ideas on the alias. You are saying that all wrappers are alias, aren't you?

Reply:
If you are just delegating to the "correct"/new implementation, then that sounds like an alias to me. If we are maintaining large parallel/duplication implementation, then that is a different story.